mass_mailing_unique/models/mass_mailing.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2015 Grupo ESOC Ingeniería de Servicios, S.L.U. - Jairo Llopis
+# Copyright 2016 Tecnativa - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import models, api, _
+from odoo.exceptions import ValidationError
+
+
+class MailMassMailingContact(models.Model):
+    _inherit = "mail.mass_mailing.contact"
+
+    @api.constrains('email', 'list_ids')
+    def _check_email_list_ids(self):
+        for contact in self:
+            other_contact = self.search([
+                ('email', '=', contact.email),
+                ('id', '!=', contact.id)
+            ])
+            if contact.list_ids & other_contact.mapped('list_ids'):
+                raise ValidationError(_("Cannot have the same email more "
+                                        "than once in the same list"))
+
+
+class MailMassMailingList(models.Model):
+    _inherit = "mail.mass_mailing.list"
+    _sql_constraints = [
+        ("unique_name", "UNIQUE(name)",
```

Comment:
Same problem is going to happen here with case variations. For instance:

```
- My list of contacts
- my list of contacts
- My List Of Contacts
- MY LIST OF CONTACTS
...
```
They all are going to pass over the constrain.

Reply:
With postgre, i think you can not make an insensitive unique constraint to not distinguish between uppercase and lowercase. To solve that, we can make an UNIQUE INDEX, for example:
CREATE UNIQUE INDEX "name_unique_idx" ON "mail_mass_mailing_list" (LOWER ('name'))
But then we must catch an IntegrityError exception in the create and write method or in another place. Actually, I do not know if it's worth doing all of this for this specific case

Reply:
Fortunately it's possible to add a constraint case insensitive. Just put `UNIQUE (LOWER(email))`. See for example: http://shuber.io/case-insensitive-unique-constraints-in-postgres/

Reply:
In this page it says that Unfortunately, Postgres does now allow us to define unique constraint using LOWER and according to this page one of the solutions is to make a 'UNIQUE INDEX'. That's what I said above and then it can be more work

Reply:
Right, I read it bad :(

OK, let's keep it that way. @chienandalu please give your final review